### PR TITLE
Get and show data list categories

### PR DESCRIPTION
**Pull request by ghost**
## Description
Get and show data list categories
## Reference Requirements
- Product Backlog ID: PBI-6
- Task ID: #6.2

## Reference Isuees
#{issue_id}

## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

- config/define.php

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- select `categories`.`id`, `categories`.`title`, COUNT(DISTINCT(books.id)) AS total_books from `categories` left join `books` on `books`.`category_id` = `categories`.`id` group by `categories`.`id` limit 10 offset 0
image

### CODE REVIEW

File: resources/views/backend/categories/index.blade.php [hunk 1]
```diff
@@ -37,10 +37,11 @@
                 </tr>
                 </thead>
                 <tbody>
-                  <tr>
-                    <td class="text-center">1</td>
-                    <td>Ea laboriosam eum in numquam tempor labore distinctio Eos quibusdam</td>
-                    <td class="text-center">41</td>
+                @foreach ($categories as $index => $category)
+                  <tr class="item-{{ $category->id }}">
```
Comment (at-caonguyen) [inline resources/views/backend/categories/index.blade.php]: Vì sao class phải khác tên nhau

File: app/Http/Controllers/Admin/CategoryController.php [hunk 2]
```diff
@@ -14,6 +17,16 @@ class CategoryController extends Controller
      */
     public function index()
     {
-        return view('backend.categories.index');
+        $fields = [
+            'categories.id',
+            'categories.title',
+            DB::raw('COUNT(DISTINCT(books.id)) AS total_books'),
```
Comment (at-caonguyen) [inline app/Http/Controllers/Admin/CategoryController.php]: Nên dùng `distinct` không vậy

**Comment by at-caonguyen** [comment]
LGTM :+1:

File: app/Http/Controllers/Admin/CategoryController.php [hunk 3]
```diff
@@ -14,6 +17,16 @@ class CategoryController extends Controller
...
+        ];
+        $categories = Category::select($fields)
+        ->leftJoin('books', 'books.category_id', '=', 'categories.id')
```
Comment (at-huenguyen) [inline app/Http/Controllers/Admin/CategoryController.php]: lùi vào 4 spaces nha cưng :smile:

Comment (at-nhannguyen2) [reply app/Http/Controllers/Admin/CategoryController.php]: :joy:

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**